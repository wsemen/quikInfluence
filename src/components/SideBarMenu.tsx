import { Flex, Box } from '@chakra-ui/layout';
import React, { useEffect, useState } from 'react';
import quikColorConstants, { sidebarBg } from 'utils/constants/colorConstants';
import NextLink from './NextLink';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { SideBarMenuOptions } from 'modules';
import { SideBarOptionMenu } from 'types';
import { useRouter } from 'next/router';
import { ColorMode } from '@chakra-ui/react';

interface SideBarMenuProps {
  bgColor?: string;
  color?: string;
  colorMode: ColorMode;
}

const SideBarMenu = ({ bgColor, color, colorMode }: SideBarMenuProps) => {
  const [activeMenu, setActiveMenu] = useState('');
  const route = useRouter();
  const { pathname } = route || { pathname: '/' };
  const _sideBarOptions = Object.values(SideBarMenuOptions);

  useEffect(() => {
    if (!activeMenu) {
      setActiveMenu(_sideBarOptions[0].path);
    }
  }, [_sideBarOptions, activeMenu]);

  return (
    <Flex flexDirection="column" width="250px" py={10} bgColor={bgColor}>
      {_sideBarOptions.map(
        ({ name, icon, path, isShown }: SideBarOptionMenu) => {
          if (!isShown) return;

          return (
            <Box
              key={name}
              py={5}
              px={10}
              minW="100%"
              _hover={{
                backgroundColor: sidebarBg[colorMode],
              }}
              bg={
                (activeMenu || pathname) === path
                  ? sidebarBg[colorMode]
                  : 'transparent'
              }
              borderLeft={`3px solid ${
                (activeMenu || pathname) === path
                  ? quikColorConstants.influenceRed
                  : sidebarBg[colorMode]
              }`}
              onClick={() => setActiveMenu(path)}
            >
              <NextLink
                href={path}
                display="block"
                _hover={{
                  textDecoration: 'none',
                  color: 'inherit',
                }}
                _focus={{
                  border: 'none',
                  textDecoration: 'none',
                }}
                fontFamily="Avenir"
                fontWeight="bold"
              >
                <FontAwesomeIcon icon={icon} /> {name}
              </NextLink>
            </Box>
          );
        }
      )}
    </Flex>
  );
};

export default SideBarMenu;
