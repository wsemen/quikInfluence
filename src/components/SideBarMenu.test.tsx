import { render } from '@testing-library/react';
import SideBarMenu from './SideBarMenu';

it('renders SideBarMenu component', () => {
  const { container } = render(<SideBarMenu colorMode="dark" />);
  expect(container).toMatchSnapshot();
});
