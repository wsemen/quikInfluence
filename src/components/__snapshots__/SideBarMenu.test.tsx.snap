// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SideBarMenu component 1`] = `
<div>
  <div
    class="css-1rzqydi"
  >
    <div
      class="css-ryypga"
    >
      <a
        class="chakra-link css-1e6bsnr"
        href="/dashboard"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-grip-horizontal fa-w-14 "
          data-icon="grip-horizontal"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 288H32c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm160 0h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm160 0h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zM96 96H32c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm160 0h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm160 0h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32z"
            fill="currentColor"
          />
        </svg>
         
        Overview
      </a>
    </div>
  </div>
</div>
`;
